File: frontend/src/api/course.ts
import api from './index'

export interface Course {
  id: number
  name: string
  type: string
  unitPrice: number
  trialPrice: number
  status: string
  createdAt: string
}

export const courseApi = {
  getAll: () => api.get('/courses'),
  getActive: () => api.get('/courses/active'),
  getById: (id: number) => api.get(`/courses/${id}`),
  create: (data: Partial<Course>) => api.post('/courses', data),
  update: (id: number, data: Partial<Course>) => api.put(`/courses/${id}`, data),
  delete: (id: number) => api.delete(`/courses/${id}`)
}

File: backend/src/main/java/com/edu/management/entity/Course.java
package com.edu.management.entity;

import com.edu.management.enums.CourseStatus;
import com.edu.management.enums.CourseType;
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "courses")
public class Course {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false, length = 100)
    private String name;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private CourseType type;
    
    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal unitPrice;
    
    // 试课价格（通常为0或体验价）
    @Column(precision = 10, scale = 2)
    private BigDecimal trialPrice = BigDecimal.ZERO;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private CourseStatus status = CourseStatus.ACTIVE;
    
    @CreationTimestamp
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    private LocalDateTime updatedAt;
}
